stan/math/prim/mat/fun/cholesky_decompose_gpu.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef STAN_MATH_PRIM_MAT_FUN_CHOLESKY_DECOMPOSE_GPU_HPP
```

Comment:
You'll notice that `prim/cholesky_decompose.cpp` is not in this PR. With `-D STAN_GPU` I was unable to get that piece of code to work even with `disable_if()`.

During compilation for the `fvar` tests there would be a failure because the ViennaCL code did not understand how to compile wrt `fvar`

Reply:
This is another reason I think it would be good to have separate gpu functions for now. I'm rather new to templating, but I'll browse around to see if there is a solution I missed wrt this.